pkg/storage-init/storage-init.sh

```diff
@@ -34,6 +34,57 @@ fi
 
 P3_FS_TYPE="ext3"
 FSCK_FAILED=0
+
+# First lets see if we're missing P3 altogether and try to create it.
+# This is safe, since the worst case scenario we may hose the system,
+# but the system without P3 is a warm brick anyways
```

Comment:
What about the case when IMGB is fine, and eveimage update places a bad version on IMGA which incorrectly decides to reformat the disk, thereby blowing away IMGB, and then the watchdog fires and tries to cause a fallback to the now gone IMGB version?

BTW: Why is this code not symmetric with respect to IMGA/IMGB? IMGA is not special.